app/src/main/java/com/kms/cura/view/fragment/RegDoc_ProfessionalFragment.java

```diff
@@ -215,4 +247,45 @@ public void callBackUpdateSpinner() {
         spnSpeciality.setAdapter(adapterSpeciality);
         spnSpeciality.setSelection(adapterSpeciality.getCount());
     }
+
+    @Override
+    public void handleEvent(String event, String data) {
+        switch (event) {
+            case EventConstant.GET_ALL_SPECIALITY_NAME_SUCCESS:
+                speciality = (ArrayList<String>) DataConstant.getInstance().getAllSpecialityName();
+                doneRequestSpeciality = true;
+                break;
+            case EventConstant.GET_ALL_FACILITY_NAME_SUCCESS:
+                facility = (ArrayList<String>) DataConstant.getInstance().getAllFacilityName();
+                doneRequestFacility = true;
+                SpecialityController.getAllSpeciality();
+                break;
+            case EventConstant.GET_ALL_DEGREE_NAME_SUCCESS:
+                degree = (ArrayList<String>) DataConstant.getInstance().getAllDegreeName();
+                doneRequestDegree = true;
+                FacilityController.getAllFacility();
+                break;
+            case EventConstant.CONNECTION_ERROR:
+                ErrorController.showDialog(getContext(), getResources().getString(R.string.ConnectionError));
+                break;
+            case EventConstant.GET_ALL_FAILED:
+                String internalError = getResources().getString(R.string.InternalError);
+                ErrorController.showDialog(getContext(),internalError + data);
+
+        }
+        if(doneRequestSpeciality && doneRequestDegree && doneRequestFacility ){
```

Comment:
doneRequestSpeciality, doneRequestDegree, doneRequestFacility should be local variables, not fields